dispatch/static/manager/src/js/pages/EventAudit/index.js

```diff
@@ -0,0 +1,181 @@
+import React from 'react'
+import R from 'ramda'
+import ReactCSSTransitionGroup from 'react-addons-css-transition-group'
+import { connect } from 'react-redux'
+import DocumentTitle from 'react-document-title'
+import { replace } from 'react-router-redux'
+
+import eventsActions from '../../actions/EventsActions'
+
+import EventCard from './EventCard'
+import EventAuditPagination from './EventAuditPagination'
+
+require('../../../styles/components/event_audit.scss')
+
+const PER_PAGE = 15
+
+const PENDING_QUERY = {
+  pending: 1,
+  limit: PER_PAGE
+}
+
+class EventAuditPage extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      count: 0
+    }
+  }
+
+  getQuery(props) {
+    props = props || this.props
+
+    const query = R.clone(PENDING_QUERY)
+
+    if (props.location.query.page) {
+      query.offset = (Number(props.location.query.page) - 1) * PER_PAGE
+    }
+
+    return query
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.events.count != this.props.events.count) {
+      this.setState({ count: nextProps.events.count })
+    }
+
+    if (nextProps.events.count && !this.eventCount
+        && !this.reloadListFlag) {
+      this.props.listEvents(this.props.token, this.getQuery(nextProps))
+      this.reloadListFlag = true
+    } else if (this.props.location.query.page != nextProps.location.query.page){
+      this.props.listEvents(this.props.token, this.getQuery(nextProps))
+    }
+
+  }
+
+  componentWillMount() {
+    this.props.listEvents(this.props.token, this.getQuery())
+
+    if (this.props.events) {
+      this.setState({ count: this.props.events.count })
+    }
+  }
+
+  checkPage() {
+    if (this.eventCount == 1 && this.props.location.query.page > 1) {
+      let query = R.clone(this.props.location.query)
+
+      query.page--
+
+      this.props.replace({
+        pathname: this.props.location.pathname,
+        query: query
+      })
+    }
+  }
+
+  approve(id) {
+    const event = this.props.entities.events[id]
+    event.is_submission = 0
+    this.props.saveEvent(this.props.token, id, event)
+    this.checkPage()
+    this.setState({ count: this.state.count - 1 })
+    this.reloadListFlag = false
+  }
```

Comment:
I think it might be better to create a new action for this, and then just call `this.props.approveEvent(this.props.token, id)`